src/redux/features/observations/plantingSiteDetailsSelectors.ts

```diff
@@ -94,15 +94,9 @@ export const selectPlantingSiteZones = createSelector(
           plantingSubzones: zone.plantingSubzones.map((sz) => ({
             ...sz,
             // generate monitoring plots from book-keeping data
-            monitoringPlots: zone.plantingSubzones
```

Comment:
this was the bug, populating monitoring plots for all subzones (note that the context here is specific to a subzone, we only need to populate monitoring plots for the subzone in context)